### DISCUSSION TITLE
service call to control "climate/set_temperature"

### DISCUSSION THREAD

**Discussion by marekchowaniok** [jmcollin78/solar_optimizer#26] [General]
Hi @jmcollin78 , 

I came across this repo/HACS and I wonder if it is possible to make service call with payload. 

For example - I have heating system, which is not standard switch but it is climate entity. To turn on heating I would raise temperature and to turn off heating I would decrease temperature. 

It is possible to do it through homeassistant UI: 

![image](https://github.com/jmcollin78/solar_optimizer/assets/404424/3f6ecaf8-a2a4-48ab-a9f3-97696abf9d6e)

If we can't call anything else, do you have any recommendation how to create "switch helper" which would raise/drop temperature on heating ? 
Or any other recommendation how would/did you solve this ? 

thanks
Marek

**Comment by jmcollin78** [comment]
I will keep this idea for a future use. I have done something similar in Versatile Thermostat (for controling a central boiler). So it is quite easy to do and can be useful. Thanks for report !

**Reply by marekchowaniok** [reply]
Do you have any example how to make "switch helper" which would make another service call ?

**Reply by marekchowaniok** [reply]
I found example how to create switch for another service calls

as an example, this might be nice starting point
https://community.home-assistant.io/t/smartir-control-your-climate-tv-and-fan-devices-via-ir-rf-controllers/100798/1334

**Reply by jmcollin78** [reply]
You have also an example in the README (search Heatzy)
